packages/material-ui/src/TextField/TextField.js

```diff
@@ -67,6 +67,7 @@ const TextField = React.forwardRef(function TextField(props, ref) {
     FormHelperTextProps,
     fullWidth,
     helperText,
+    hiddenLabel,
```

Comment:
Never seen the first usage. A hidden `label` will penalized by lighthouse though it will be announced by chrome vox. Will look at nvda as well but I would prefer an easy a11y API. Requiring `<TextField inputProps={{ 'aria-label': 'The label' }} />` isn't in the spirit of "the obvious API should result in the better result".

Reply:
Sorry, wrong example.

Reply:
Updated

Reply:
Should we update all the demos?

Reply:
What pattern and why? what issue are you trying to solve?

Reply:
The issue I'm trying to avoid is code fragmentation. I think that we should avoid to encourage two alternative APIs for the same task. If we can avoid people to wonder which one is the best approach, I think that it will be simpler for them :).

Reply:
Those are two different things though. The first makes only sense for margin="dense". The second one creates no `label` element. The second approach also has implicit stylistic behavior for filled and margin which isn't good either. 

So no these are not the same thing. Especially considering this is new behavior. You're actually constructing issues here.

Reply:
Also wondering where we "encourage" two alternative APIs. Where are we doing this currently for this new behavior.

Reply:
By encourage, I mean something we document. 

Then I don't understand the global picture. More specifically, I don't understand the following points:
- why we are rendering a label to then hide it? Can we use arial-label exclusively without mounting the label at the TextField level? 
- why is the `hiddenLabel` prop not usable with all the margin values? 
- why is the `hiddenLabel` prop marked as @ignore in the prop types? Should we add a description?

Reply:
> why is the hiddenLabel prop marked as @ignore in the prop types? Should we add a description?

Just for consistency with the other form control props but I'm open to remove it. It should come from the context anyway (as all the other props but I focused on continuing the current implementation to reduce friction but it seems like you seek it).

> why is the hiddenLabel prop not usable with all the margin values?

There's no mock for other input variants with hidden labels. You'd have to ask the Material designers.

> why we are rendering a label to then hide it? Can we use arial-label exclusively without mounting the label at the TextField level?

I was hoping that it would still be considered for screenreaders (which is the case for chrome vox as I said. aria-label is just as a fallback used here). As far as I know React already uses `hidden` to deprioritize those elements. What is the concern here other then "render counting"?

Reply:
> By encourage, I mean something we document.

That's what I don't want. If I forgot to remove it from the docs please add a comment to that line. As far as I can tell we only document a single approach.

Reply:
> Just for consistency with the other form control props

This sounds like a great reason.

> There's no mock for other input variants with hidden labels. You'd have to ask the Material designers.

Would hiddenLabel work with the outlined and standard `variant` prop?

> That's what I don't want.

Then the wording of "encourage", from my part, was too strong. Maybe we could try to replace all the arial-label demos with hiddenLabel?

Reply:
I was working on the Rating this morning, https://www.w3.org/WAI/tutorials/forms/custom-controls/ use some custom CSS to hide the label from the users.

Reply:
Somebody can work on this. I just wanted to implement a single variant. I would never encourage anybody to use inputs without visible labels but since it's part of the Material guidelines and you marked it as important I implemented it. Beyond that I don't want to proliferate IMO harmful UX. `label + hiddenLabel` looks too clean. `inputProps={{ 'aria-label': label }}` is intended to be an eye-sore exactly because of this.

Reply:
> uses custom style to hide the label from the users (.visuallyhidden).

Yeah we can probably trick screen readers into considering `label` in a `<FormControl hiddenLabel />`. I just want to avoid an arms-race between smarter screen readers (consider dimensions, position, visbility, transitions etc) and implementors by using `aria-label` as a fallback.

Reply:
> inputProps={{ 'aria-label': label }} is intended to be an eye-sore exactly because of this.

Do you mean that you prefer this API over `label + hiddenLabel` because I encourage the usage of a visual label? I agree with the reasoning. In this case, I would propose that we don't support `label + hiddenLabel` in the first place. We could only support `hiddenLabel + inputProps={{ 'aria-label': label }}` to star with. What do you think? It's the last important concern I have with the pull request, otherwise 👍.

Reply:
Well no because `label + hiddenLabel` is fine for `variant="filled" margin="dense"` as per your decree. I don't want supported behavior to be bad. But it shouldn't be used for everything.

Reply:
> Well no because label + hiddenLabel is fine for variant="filled" margin="dense" as per your decree.

I don't remember saying that. What if we remove my bias from the equation, does it change the result?